src/common/SSLUtil.cc

```diff
@@ -29,61 +28,59 @@ namespace ssl_util
 
     void base64_decode(const std::string& in, std::string& out)
     {
-        BIO * bio_64  = BIO_new(BIO_f_base64());
+        const int max_size = 3 * in.length()/4 + 1;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that an integer overflow arising for your program. It happens when the outcome of an arithmetic operation surpasses the maximum representable value for the utilized data type. It can lead to unpredictable behavior, security vulnerabilities, and system crashes. To address this vulnerability, it is essential to perform thorough range checking and validation of input values. Employ data types with adequate size to accommodate the anticipated range of values and implement checks to identify and manage potential overflow conditions.